### DISCUSSION TITLE
Fmincon vs Ipopt

### DISCUSSION THREAD

**Discussion by hishamop** [coin-or/Ipopt#484] [Q&A] [answered]
I'm trying to solve a nonlinear convex minimization problem with linear constraints which I have 
solved successfully using MATLAB fmincon.

In fmincon, I used the  option configuration  as below
**options=optimset('Algorithm','sqp','ScaleProblem','obj-and-constr','DerivativeCheck','off','GradObj','on','TolX',1e-16,'TolFun',1e-16,'MaxFunEvals',8000,'MaxIter',6000);**
and I got local minima after 465 iterations.

In Ipopt, I have modelled my objective function, constraints and its gradients without errors, but couldn't solve the problem. I used various linear solvers and scaling. The magnitude of variables in the problem ranges from 10^-3 to 10^3 and I need to get the value of local minima below 10^-9. 

The problem doesn't converge to local minima even after 200000 iterations.
Even after I used the solution which I got from matlab as initial values, Ipopt still iterates and doesn't converge.

Since I am not from a mathematical background I don't have much knowledge about Ipopt options.
Could anyone help me please.

**Comment by svigerske** [comment, marked as answer]
Without anything to look at or something to reproduce what you see, there is nothing we can do.
